torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1615,7 +1608,10 @@ std::vector<Val*> Index::getGlobalProducerStridedIndices(
         loops,
         root_ind);
 
-    root_ind = getProducerIndexWithHalo(producer_tv, i, root_ind, consumer_tv);
+    if (!override_index.count(root_dom[i])) {
```

Comment:
Added a comment the original comment. I think we should remove the assertion and also this `if`. It looks inconsistent to have a predicate only for `getProducerIndexWithHalo`, whereas `getProducerIndexWithGather`, etc. are just called always.

Reply:
changed